### PR TITLE
[SDK][Bot-Dialogs] Update Recognizers-Text internal library

### PR DESCRIPTION
**Pull request by Batta32**
Fixes 900
Fixes 901

## Description
The [Recognizers-Text](https://github.com/microsoft/Recognizers-Text) repository was updated with new changes for the Java library, so the internal library can be updated with these changes.

## Specific Changes
- Update `Choice` library
- Update `Datetime` library
- Update `Number` library
- Update `NumberWithUnit` library
- Update `Main` library
- Add `Expression` library
- Add `Sequence` library
- Remove [deprecated workaround](https://github.com/microsoft/botbuilder-java/blob/main/libraries/bot-dialogs/src/main/java/com/microsoft/bot/dialogs/choices/ChoiceRecognizers.java#L134) as it's already fixed in Recognizers-Text (see microsoft/Recognizers-Text2461)

We manually updated the resources of DateTime as the build in Java fails because the resources are not updated. We created the microsoft/Recognizers-Text2508 to apply those changes in Recognizers-Text repository too.

## Testing

_Internal Recognizers-Text structure updated_
image

_mvn clean install passing correctly_
image

### CODE REVIEW

**Review by fran893 — APPROVED** [review]

File: libraries/bot-dialogs/src/main/java/com/microsoft/recognizers/text/datetime/resources/EnglishDateTime.java [hunk 1]
```diff
@@ -1231,8 +1231,11 @@
 
     public static final ImmutableMap<String, Integer> DayOfMonth = ImmutableMap.<String, Integer>builder()
         .put("1st", 1)
+        .put("1th", 1)
```
Comment (matiasroldan6) [inline libraries/bot-dialogs/src/main/java/com/microsoft/recognizers/text/datetime/resources/EnglishDateTime.java:1234]: This is strange but I assume these additions (..th) are for a reason, right?

Comment (Batta32) [reply libraries/bot-dialogs/src/main/java/com/microsoft/recognizers/text/datetime/resources/EnglishDateTime.java:1234]: I can't tell about this as it's a copy paste from RT. However, the PR which includes this in RT justify this as: `Support to recognize date like "1th", "2th" in all platforms.`

File: libraries/bot-dialogs/src/main/java/com/microsoft/recognizers/text/number/resources/GermanNumeric.java [hunk 2]
```diff
@@ -166,12 +168,107 @@ public static String DoubleWithoutIntegralRegex(String placeholder) {
     public static final String DoubleAllFloatRegex = "((?<=\\b){AllFloatRegex}(?=\\b))"
             .replace("{AllFloatRegex}", AllFloatRegex);
 
+    public static final String ConnectorRegex = "(?<spacer>und)";
+
     public static final String NumberWithSuffixPercentage = "(?<!%)({BaseNumbers.NumberReplaceToken})(\\s*)(%(?!{BaseNumbers.NumberReplaceToken})|prozent(punkte)?\\b)"
             .replace("{BaseNumbers.NumberReplaceToken}", BaseNumbers.NumberReplaceToken);
 
-    public static final String NumberWithPrefixPercentage = "(Prozent)(\\s*)({BaseNumbers.NumberReplaceToken})"
+    public static final String NumberWithPrefixPercentage = "(prozent)(\\s*)({BaseNumbers.NumberReplaceToken})"
```
Comment (matiasroldan6) [inline libraries/bot-dialogs/src/main/java/com/microsoft/recognizers/text/number/resources/GermanNumeric.java:176]: all nouns in german start with capital letter. I guess this change is ok though

**Review by FedericoBernal — APPROVED** [review]

**Review by matiasroldan6 — APPROVED** [review]

**Comment by Batta32** [comment]
Merged in MS

**Pull request closed without merging**